### PR TITLE
gb2697 fix pagination

### PR DESCRIPTION
**Pull request by Finistere**
- Add Nix Flake for dev shell (linux)
- fix(cli/tests): Fix Pagination tests

# Description

Add Nix for local development shell (linux-only for now!). 
Fix pagination tests.

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [X] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

Fixes: GB-2697

### CODE REVIEW

File: cli/crates/cli/tests/graphql/pagination/paginate-todo-lists.graphql [hunk 1]
```diff
@@ -1,15 +1,21 @@
 query PaginateTodoLists($first: Int, $last: Int, $before: String, $after: String) {
-  todoListCollection(first: $first, last: $last, before: $before, after: $after) {
```
Comment (yoav-lavi) [inline cli/crates/cli/tests/graphql/pagination/paginate-todo-lists.graphql:2]: Why remove the arguments here? Makes sense to leave it as a possibility

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
```
Comment (yoav-lavi) [inline flake.nix:1]: Could we split the flake changes into a separate PR?

File: cli/crates/cli/tests/utils/macros.rs [hunk 3]
```diff
@@ -2,22 +2,24 @@ pub use json_dotpath::DotPaths;
 
 #[macro_export]
 macro_rules! dot_get_opt {
-    ($item: ident, $dotpath: literal, $ty: ty) => {{
+    ($item: expr, $dotpath: expr, $ty: ty) => {{
```
Comment (yoav-lavi) [inline cli/crates/cli/tests/utils/macros.rs:5]: Why did these change?
We originally debated allowing an expression but there wasn't consensus, plus we want to keep parity with the API version.

We could discuss this again

File: cli/crates/cli/tests/utils/macros.rs [hunk 4]
```diff
@@ -2,22 +2,24 @@ pub use json_dotpath::DotPaths;
...
-    ($item: ident, $dotpath: literal) => {{
+    ($item: expr, $dotpath: expr) => {{
         use json_dotpath::DotPaths;
         $item.dot_get::<_>($dotpath).ok().flatten()
     }};
 }
 
 #[macro_export]
 macro_rules! dot_get {
-    ($item: ident, $dotpath: literal, $ty: ty) => {{
```
Comment (yoav-lavi) [inline cli/crates/cli/tests/utils/macros.rs:17]: Why did these change?

File: cli/crates/cli/tests/graphql/pagination/paginate-todo-lists.graphql (hunk 1 above)
Comment (Finistere) [reply cli/crates/cli/tests/graphql/pagination/paginate-todo-lists.graphql:2]: Cursors aren't the same so I removed them. I kept `first` & `last` because they could eventually help find a bug and it's free even though it should be doing the same thing.
Otherwise adding specific cursors wouldn't bring that much IMHO. We're already testing root pagination. We could also test root pagination in conjunction with nested one, but that's not what we're doing currently. Should we add it eventually, we could just add the parameters then.

File: flake.nix (hunk 2 above)
Comment (Finistere) [reply flake.nix:1]: Yeah, had it lying around and got lazy. Will separate it. :)

File: cli/crates/cli/tests/utils/macros.rs (hunk 3 above)
Comment (Finistere) [reply cli/crates/cli/tests/utils/macros.rs:5]: Because literal doesn't work with the `format!("{prefix}.edges...")` that I'm using in my loop to test nested & root pagination. It stays consistent in terms of usage with the API, a literal is still an expression.

If I remove it, I have to unroll my loop. So I'm not sure to see any benefit to not support expression currently. ¯\\_(ツ)_/¯

File: flake.nix (hunk 2 above)
Comment (Finistere) [reply flake.nix:1]: Well and its was in a different commit, so it felt OK to me. But I have no problem creating another PR.

File: cli/crates/cli/tests/utils/macros.rs (hunk 3 above)
Comment (yoav-lavi) [reply cli/crates/cli/tests/utils/macros.rs:5]: I see, ok let's change it then

**Review by yoav-lavi — APPROVED** [review]
Looks good!

File: cli/crates/cli/tests/pagination.rs [hunk 5]
```diff
@@ -4,301 +4,171 @@ mod utils;
...
-    PAGINATION_MUTATION, PAGINATION_PAGINATE_TODOS, PAGINATION_PAGINATE_TODO_LISTS, PAGINATION_SCHEMA,
+    PAGINATION_CREATE_TODO, PAGINATION_CREATE_TODO_LIST, PAGINATION_PAGINATE_TODOS, PAGINATION_PAGINATE_TODO_LISTS,
+    PAGINATION_SCHEMA,
 };
 use utils::environment::Environment;
 
 #[test]
-
-fn pagination() {
+fn test_pagination() {
```
Comment (yoav-lavi) [inline cli/crates/cli/tests/pagination.rs:13]: Why was the name here changed? This makes the output a bit less clean since it shows up in certain locations (and you're running tests to begin with)

Comment (Finistere) [reply cli/crates/cli/tests/pagination.rs:13]: Ah force of habit, changing it

**Comment by yoav-lavi** [comment]
@Finistere can we merge this?

**Comment by Finistere** [comment]
Yes
